Fixes and Updates

gamecoder95 opened the pull request:
- User can now successfully deregister
- Added static methods to the REASON enum to be able to obtain the text values (before it was only displaying the binary string "key") - response messages are correctly displayed now, and made them more user-friendly.
- A client's own offers don't show up as items he can bid on.
- Fixed the item number display and removed the hack in the bid callback.
- When deregistering successfully, all input and output fields on the GUI are cleared.

gamecoder95 committed:
Fixed dereg conf bug

Was occurring because the callback was taking more parameters than was
needed.

samratdebroy commented on gui/auction_client_gui.py:
```diff
@@ -129,6 +129,13 @@ def activate_client_interface(self, activate):
 
         self.set_disabled(self.reg_panel, activate)
 
+        if not activate:
```
I actually think it might make more sense to clear upon receiving a Registered Message. You want to still be able to look at the your closing state after deregistering. You just need to clear the item lists when a new client session begins, so after a new register.

gamecoder95 replied:
Done, and in addition, the new offer input fields never get cleared as there's no harm in that.

samratdebroy reviewed (changes requested):
Just one change, I think we should only clear GUI after a register

gamecoder95 committed:
Fixed reason strings

The reason enum string values weren't accessible. Created static methods
in the REASON enum to be able to retrive them, formatted the client code
to add that and display those in the GUI.

gamecoder95 committed:
GUI doesn't add client's offered items to the 'New Items' list

samratdebroy reviewed (approved):
looks good to me

gamecoder95 committed:
Server sends new client all of the items offered

gamecoder95 committed:
Deregistering clears GUI

All lists and user input and output labels are cleared on a successful
deregister.

gamecoder95 committed:
Merge branch 'dereg_fix' of github.com:samratdebroy/AuctionSystem into dereg_fix

gamecoder95 committed:
Removed the send_one_all_items function

gamecoder95 committed:
Fixed bid callback to get the value from the correct entry.

gamecoder95 committed:
Corrected offer info panel to display item...

...properly.

gamecoder95 committed:
Implemented changes

GUI is now disabled upon registration; the entry fields for creating a new offer
never get cleared.
